piff/gp_interp.py

```diff
@@ -173,42 +216,84 @@ def interpolateList(self, stars, logger=None):
+                for j in range(self.nparams):
+                    y0_updated[self.rows[j]] = y0[j] 
+                fit = star.fit.newParams(y0_updated)
             fitted_stars.append(Star(star.data, fit))
         return fitted_stars
 
     def _finish_write(self, fits, extname):
         # Note, we're only storing the training data and hyperparameters here, which means the
         # Cholesky decomposition will have to be re-computed when this object is read back from
         # disk.
-        init_theta = self.gp.kernel.theta
-        fit_theta = self.gp.kernel_.theta
+        init_theta = np.array([self._init_theta[i] for i in range(self.nparams)])
+        fit_theta = np.array([ker.theta for ker in self.kernels])
         dtypes = [('INIT_THETA', init_theta.dtype, init_theta.shape),
                   ('FIT_THETA', fit_theta.dtype, fit_theta.shape),
                   ('X', self._X.dtype, self._X.shape),
-                  ('Y', self._y.dtype, self._y.shape)]
+                  ('Y', self._y.dtype, self._y.shape),
+                  ('Y_ERR', self._y_err.dtype, self._y_err.shape),
+                  ('ROWS', self.rows.dtype,  self.rows.shape),
+                  ('OPTIMIZER', str, len(self.optimizer))]
 
         data = np.empty(1, dtype=dtypes)
         data['INIT_THETA'] = init_theta
         data['FIT_THETA'] = fit_theta
         data['X'] = self._X
         data['Y'] = self._y
+        data['Y_ERR'] = self._y_err
+        data['ROWS'] = self.rows
+        data['OPTIMIZER'] = self.optimizer
 
         fits.write_table(data, extname=extname+'_kernel')
 
     def _finish_read(self, fits, extname):
         data = fits[extname+'_kernel'].read()
-        # Run fit to set up GP, but don't actually do any hyperparameter optimization.  Just
+        # Run fit to set up GP, but don't actually do any hyperparameter optimization. Just
         # set the GP up using the current hyperparameters.
-        self.gp.kernel.theta = np.atleast_1d(data['FIT_THETA'][0])
-        old_optimizer, self.gp.optimizer = self.gp.optimizer, None
-        self._fit(data['X'][0], data['Y'][0])
-        self.gp.optimizer = old_optimizer
-        # Now that gp is setup, we can restore it's initial kernel.
-        self.gp.kernel.theta = np.atleast_1d(data['INIT_THETA'][0])
+        # Need to give back average fits files if needed.
+
+        init_theta = np.atleast_1d(data['INIT_THETA'][0])
+        fit_theta = np.atleast_1d(data['FIT_THETA'][0])
+
+        self._X = np.atleast_1d(data['X'][0])
+        self._y = np.atleast_1d(data['Y'][0])
+        self._y_err = np.atleast_1d(data['Y_ERR'][0])
+        self.rows = np.atleast_1d(data['ROWS'][0])
+
+        self._init_theta = init_theta
+        self.nparams = len(init_theta)
+        self.optimizer = data['OPTIMIZER'][0]
+
+        if len(self.kernel_template)==1:
+            self.kernels = [copy.deepcopy(self.kernel_template[0]) for i in range(self.nparams)]
+        else:
```

Comment:
Same here.  I think this branch is impossible now.

Reply:
Same here it can still happened